### ISSUE TITLE
Error connecting to Unifi network application

### ISSUE THREAD

**Issue by nathen418** [ufozone/ha-unifi-voucher#69]
Tried everything I could think of here and I still cant get the hotspot integration  to connect to my network controller

Facts:
- I have a UDM-SE and the wifi service account has full permissions on the unifi application. ([Figure 1](https://github.com/ufozone/ha-unifi-voucher/assets/23101313/2188b45d-3041-4eb9-84c5-c85e62352352))
- The host is reachable, and no firewall fules are blocking access to the UDM from my trusted server network. I can tell this because unifi shows the last login date as 1 minute ago. ([Figure 2](https://github.com/ufozone/ha-unifi-voucher/assets/23101313/75d53ab6-293c-4dc0-89d1-a094d4de557e))
- I am running the latest version 1.2.1.
- I am running these versions of hass and hass OS
- I am running hass behind Nginx Proxy Manager but have tried this without using NPM
```
Core 2024.5.4
Supervisor 2024.05.1
Operating System 12.3.rc2
Frontend 20240501.1
```
- I get the following error on my screen and the following text in the logs:
![image](https://github.com/ufozone/ha-unifi-voucher/assets/23101313/59bdc150-f990-40e0-8b2a-8a2a81381756)
```
Logger: custom_components.unifi_voucher
Source: custom_components/unifi_voucher/api.py:142
integration: unifi_voucher ([documentation](https://github.com/ufozone/ha-unifi-voucher), [issues](https://github.com/ufozone/ha-unifi-voucher/issues))
First occurred: 10:42:25 PM (4 occurrences)
Last logged: 10:45:50 PM

    Error connecting to the UniFi Network at 10.3.1.1:
    Error connecting to the UniFi Network at 10.3.1.1: Error requesting data from https://10.3.1.1:443: Cannot connect to host 10.3.1.1:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1000)')]
```

In the hass config/home-assistant.log, I am getting 401 errors when trying to access the API path '/proxy/network/sites' after relieving the full correct payload with a token from /API/auth/login and getting a `Logged in to UniFi https://someurlhere` message in the log file. I also seem to be getting hit with 403 and 429 errors despite only trying this once every 3-10 ish minutes.

I am not sure what else to try, any clarification would be helpful. Will update as I learn more.

**Comment by ufozone** [comment]
See the error message:
`[SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1000)')]`

It looks like the setting not to validate the SSL certificate is being ignored. I'll check that.

**Comment by ufozone** [comment]
Unfortunately, I cannot reproduce it. In my tries the error only occurs when I activate the SSL verify checkbox. If I keep it deactivated, the setup works.

We need more logs. Can you please activate the correct DEBUG mode:
https://github.com/ufozone/ha-unifi-voucher#debugging

And then provide me with the log from the first login attempt. Like:
```
2024-05-22 19:39:53.690 WARNING (MainThread) [aiounifi.interfaces.connectivity] Using SSL context True
2024-05-22 19:39:53.690 DEBUG (MainThread) [aiounifi.interfaces.connectivity] sending (to https://unifi:443) get, None, {'allow_redirects': False}
2024-05-22 19:39:53.711 ERROR (MainThread) [custom_components.unifi_voucher] Error connecting to the UniFi Network at unifi: Error requesting data from https://unifi:443: Cannot connect to host unifi:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1000)')]
```

**Comment by ufozone** [comment]
@nathen418 , does the default UniFi integration work for you?
https://www.home-assistant.io/integrations/unifi/

The UniFi Hotspot Integration uses the same Auth and the same asyncio library as the default UniFi integration.

**Comment by nathen418** [comment]
I will try when I get off of work today and paste my logs in here

**Comment by nathen418** [comment]
Sorry for the late response, I was finishing moving the rest of my things to a new house. Yes, the default unifi integration worked just fine, and I was able to log in with no issues.

I just installed the 1.2.2 update, though, and it seems to have fixed all of the issues, though I also updated my software versions to the following, so that may have also affected the change. I haven't modified my firewall rules at all either.

Previous versions:
```
ha-unifi-voucher: 1.2.1
Core 2024.5.4
Supervisor 2024.05.1
Operating System 12.3.rc2
Frontend 20240501.1
```

Current versions:
```
ha-unifi-voucher: 1.2.2
Core 2024.5.5
Supervisor 2024.05.1
Operating System 12.3.rc2
Frontend 20240501.1
```
